api/api.go

```diff
@@ -188,18 +194,25 @@ func (binding *SplitioAPIBinding) getSegment(segmentName string) (dtos.SegmentCh
 
 	addedMap := map[string]bool{}
 	for _, changes := range allChanges {
-		var SegmentChanges dtos.SegmentChangesDTO
-		err = mapstructure.Decode(changes, &SegmentChanges)
+		var segmentChanges dtos.SegmentChangesDTO
+
+		config := &mapstructure.DecoderConfig{TagName: "json", Result: &segmentChanges}
```

Comment:
also update when converting to `Segment` although currently it doesn't have the problem if not refactoring